### DISCUSSION TITLE
Handling file paths in development vs installed directory

### DISCUSSION THREAD

**Discussion by yogat3ch** [ThinkR-open/golem#903] [Q&A] [answered]
Hey @ColinFay & @VincentGuyader,
I'm in the midst of writing supporting functions that determine the appropriate path to a directory depending on whether the `golem` is being developed or whether it's installed as the directory structure changes once installed. 
This mostly includes handling the location of files in the `inst` directory when developing vs when installed such that file paths work in both instances. 
Though I'm thinking you or someone else has already probably encountered this issue before! Are there any existing solutions to making sure file file paths work properly in both states of the package?

**Comment by ColinFay** [comment, marked as answer]
Hey @yogat3ch, 

This is what the `app_sys()` function was designed for. You can find it inside the app_config.R file. 


```
L inst/
  L this.xyz
```

can be read with `app_sys("this.xyz")`

```
L inst/
  L that / 
    L this.xyz
```

can be read with `app_sys("that/this.xyz")` 

This function is a wrapper around `system.file(package = )`, that allows to point to the current directory (during dev) or to the path where the package is installed __on the current machine__. That allows to refer to file on any computer. 

For example, on my laptop, if I want to refer to https://github.com/ThinkR-open/golem/tree/master/inst/shinyexample

```
> system.file("shinyexample", package = "golem")
[1] "/Users/colin/Library/R/x86_64/4.2/library/golem/shinyexample
```

And this will return a different result on your machine.

**Reply by yogat3ch** [reply]
Of course ya'll already accounted for this! I did not understand the utility of the `app_sys` function somehow. Thank you for the guidance @ColinFay 🙏
